### PR TITLE
Issue/463/Added 'schrabback' option to reduced shear

### PR DESCRIPTION
**Pull request by eduardojsbarroso**

### CODE REVIEW

File: clmm/theory/func_layer.py [hunk 1]
```diff
@@ -465,7 +465,9 @@ def compute_reduced_tangential_shear(
                 `r_proj` is a float);
             * `applegate14`: use the equation (6) in Weighing the Giants - III \
                 (Applegate et al. 2014; https://arxiv.org/abs/1208.0605) to evaluate tangential reduced shear;
-
+            * `schrabback`: use the equation (12) in Cluster Mass Calibration at High Redshift \
```
Comment (shenmingfu) [inline clmm/theory/func_layer.py:468]: Do you mind using "schrabback18" as the option name (add year)? To be consistent with the other one. Thanks.

File: clmm/theory/parent_class.py [hunk 2]
```diff
@@ -332,7 +332,9 @@ def eval_reduced_tangential_shear(self, r_proj, z_cl, z_src, z_src_model='single
                     `r_proj` is a float);
                 * `applegate14`: use the equation (6) in Weighing the Giants - III \
                     (Applegate et al. 2014; https://arxiv.org/abs/1208.0605) to evaluate tangential reduced shear;
-
+                * `schrabback`: use the equation (12) in Cluster Mass Calibration at High Redshift \
```
Comment (shenmingfu) [inline clmm/theory/parent_class.py:335]: option name

File: clmm/theory/parent_class.py [hunk 3]
```diff
@@ -375,6 +377,16 @@ def eval_reduced_tangential_shear(self, r_proj, z_cl, z_src, z_src_model='single
                 gammat = self._eval_tangential_shear(r_proj, z_cl, z_source)
                 kappa = self._eval_convergence(r_proj, z_cl, z_source)
                 gt = beta_s_mean * gammat / (1. - beta_s_square_mean / beta_s_mean * kappa)
+        
+        elif z_src_model == 'schrabback':
```
Comment (shenmingfu) [inline clmm/theory/parent_class.py:381]: option name

File: tests/test_theory.py [hunk 4]
```diff
@@ -410,6 +410,11 @@ def test_shear_convergence_unittests(modeling_data):
     assert_allclose(theo.compute_reduced_tangential_shear(cosmo=cosmo, **cfg_inf['GAMMA_PARAMS'], beta_s_mean=beta_s_mean, beta_s_square_mean=beta_s_square_mean),
                     beta_s_mean * gammat_inf/(1.0 - beta_s_square_mean / beta_s_mean * kappa_inf), 1.0e-10)
 
+    cfg_inf['GAMMA_PARAMS']['z_src_model'] = 'schrabback'
```
Comment (shenmingfu) [inline tests/test_theory.py:413]: option name

File: tests/test_theory.py [hunk 5]
```diff
@@ -521,6 +526,7 @@ def test_shear_convergence_unittests(modeling_data):
     gammat_inf = mod.eval_tangential_shear(profile_pars[0], profile_pars[1], source_redshift_inf) #np.inf)
     kappa_inf = mod.eval_convergence(profile_pars[0], profile_pars[1], source_redshift_inf) #np.inf)
     assert_allclose(mod.eval_reduced_tangential_shear(*profile_pars, 'applegate14', beta_s_mean, beta_s_square_mean), beta_s_mean * gammat_inf/(1.0 - beta_s_square_mean / beta_s_mean * kappa_inf), 1.0e-10)
+    assert_allclose(mod.eval_reduced_tangential_shear(*profile_pars, 'schrabback', beta_s_mean, beta_s_square_mean), (1. + (beta_s_square_mean / (beta_s_mean * beta_s_mean) - 1.) * beta_s_mean * kappa_inf) * (beta_s_mean * gammat_inf / (1. - beta_s_mean * kappa_inf)), 1.0e-10)
```
Comment (shenmingfu) [inline tests/test_theory.py:529]: option name

File: tests/test_theory_parent.py [hunk 6]
```diff
@@ -22,6 +22,7 @@ def test_unimplemented(modeling_data):
     assert_raises(NotImplementedError, mod.eval_tangential_shear, [0.3], 0.3, 0.5)
     assert_raises(NotImplementedError, mod.eval_reduced_tangential_shear, [0.3], 0.3, 0.5)
     assert_raises(NotImplementedError, mod.eval_reduced_tangential_shear, [0.3], 0.3, 0.5, 'applegate14', 0.6, 0.4)
+    assert_raises(NotImplementedError, mod.eval_reduced_tangential_shear, [0.3], 0.3, 0.5, 'schrabback', 0.6, 0.4)
```
Comment (shenmingfu) [inline tests/test_theory_parent.py:25]: option name

File: tests/test_theory_parent.py [hunk 7]
```diff
@@ -83,3 +84,8 @@ def test_instantiate(modeling_data):
     reduced_shear = mod.eval_reduced_tangential_shear(r_proj, z_cl, np.repeat(z_src, len(r_proj)), 'applegate14', beta_s_mean, beta_s_square_mean)
     assert_allclose(reduced_shear, beta_s_mean * shear_inf/(1.0 - beta_s_square_mean / beta_s_mean * convergence_inf), rtol=1.0e-12)
     
+    reduced_shear = mod.eval_reduced_tangential_shear(r_proj, z_cl, z_src, 'schrabback', beta_s_mean, beta_s_square_mean)
```
Comment (shenmingfu) [inline tests/test_theory_parent.py:87]: option name

File: tests/test_theory_parent.py [hunk 8]
```diff
@@ -83,3 +84,8 @@ def test_instantiate(modeling_data):
...
+    assert_allclose(reduced_shear, (1. + (beta_s_square_mean / (beta_s_mean * beta_s_mean) - 1.) * beta_s_mean * convergence_inf) * (beta_s_mean * shear_inf / (1. - beta_s_mean * convergence_inf)), rtol=1.0e-12)
+
+    reduced_shear = mod.eval_reduced_tangential_shear(r_proj, z_cl, np.repeat(z_src, len(r_proj)), 'schrabback', beta_s_mean, beta_s_square_mean)
```
Comment (shenmingfu) [inline tests/test_theory_parent.py:90]: option name

**Review by shenmingfu — COMMENTED** [review]
This is great. Thank you. Just hope the option name can be changed to "schrabback18" to have the year number and to be consistent with the other one. Other things are good.

**Comment by eduardojsbarroso** [comment]
@shenmingfu  Thank you for reviewing.  I will change the option name tomorrow.  Let me know if you need something else.

**Comment by eduardojsbarroso** [comment]
Hi @combet. I think this is ready to be merged.

**Comment by shenmingfu** [comment]
Hi @combet, could you merge this PR? I'm not authorized to do it.

**Pull request merged**